scripts/extension_helpers.js

```diff
@@ -40,16 +54,51 @@ var path = require('path');
     }];
 */
 
+
 // The "always ignore" externals used by JupyterLab, Phosphor and friends
-var DEFAULT_EXTERNALS = [
-    function(context, request, callback){
-      console.log('TODO: Phosphor external rewriter...');
+var BASE_EXTERNALS = [
+    function(context, request, callback) {
+      // All phosphor imports get mangled to use the external bundle.
+      var regex = /^phosphor\/lib\/([a-z\/]+)$/;
+      if(regex.test(request)) {
+          var matches = regex.exec(request)[1];
+          var lib = 'var phosphor.' + matches.split('/').join('.');
+          return callback(null, lib);
+      }
+      callback();
     },
-    'jupyter-js-services',
-    /codemirror/
+    {
+      'jupyter-js-services': 'jupyter.services',
+      'jquery': '$',
+      'jquery-ui': '$'
+    }
   ];
 
 
+// Downstream extensions should exclude JupyterLab itself as well.
+var DEFAULT_EXTERNALS = BASE_EXTERNALS + [
+    function(context, request, callback) {
+      // JupyterLab imports get mangled to use the external bundle.
+      regex = /^jupyterlab\/lib\/([a-z\/]+)$/;
```

Comment:
nice!

On Wed, Aug 17, 2016 at 2:20 PM Jason Grout <notifications@github.com>
wrote:

> In scripts/extension_helpers.js
> <https://github.com/jupyter/jupyterlab/pull/672#discussion_r75177803>:
>
> >    ];
> >
> >
> > +// Downstream extensions should exclude JupyterLab itself as well.
> > +var DEFAULT_EXTERNALS = BASE_EXTERNALS + [
> > +    function(context, request, callback) {
> > +      // JupyterLab imports get mangled to use the external bundle.
> > +      regex = /^jupyterlab\/lib\/([a-z\/]+)$/;
>
> a-z -> \w
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jupyter/jupyterlab/pull/672/files/75f46089219f0aebda9bef07ac6574391e48220f#r75177803>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AACxRDKB3WIOq3S3UR641yVVXnYiY_gVks5qg1DwgaJpZM4JldLw>
> .
>